x-pack/test/security_solution_endpoint_api_int/apis/metadata.ts

```diff
@@ -65,11 +64,11 @@ export default function ({ getService }: FtrProviderContext) {
           ]);
 
           // wait for latest metadata transform to run
-          await new Promise((r) => setTimeout(r, 30000));
+          await new Promise((r) => setTimeout(r, 60000));
```

Comment:
The [Endpoint FTR service here](https://github.com/elastic/kibana/blob/3c54f459f55be16fb5c0ec4a9fda8468fbecf900/x-pack/test/security_solution_endpoint/services/endpoint.ts#L146) has a method that can actually wait until the actual endpoints has been loaded. Would that be a better solution than to use a `setTimeout()`?

Reply:
FYI: in that method, it uses `retry.waitFor()`, but if you needed something like that method but perhaps have control over the timeout value, we could alter it to accept a second optional argument with the timeout and have its implementation use `retry.waitForWithTimeout()` instead

Reply:
For sure. I recall we attempted this some months back but the issue was that this service was using the functional FTR provider while the integration tests weren't. Ran into some issues consolidating them and put a pin in it at the time. Definitely would be a better solution.